vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -112,6 +121,21 @@ def refresh_metrics_metadata_cache(self):
         # Apparently only when the server restarts?
         # https://pubs.vmware.com/vsphere-50/index.jsp?topic=%2Fcom.vmware.wssdk.pg.doc_50%2FPG_Ch16_Performance.18.5.html
 
+    def refresh_tags_cache(self):
+        """
+        Fetch the all tags, build tags for each monitored resources and store all of that into the tags_cache.
+        """
+        if not self.api_rest:
+            return
+        t0 = Timer()
+        try:
+            mor_tags = self.api_rest.get_resource_tags()
+        except APIConnectionError as e:
+            self.log.error("Failed to collect tags: %s", e)
+            return
+        self.histogram('datadog.vsphere.query_tags.time', t0.total(), tags=self.config.base_tags, raw=True)
```

Comment:
```suggestion
        self.gauge('datadog.vsphere.query_tags.time', t0.total(), tags=self.config.base_tags, raw=True)
```
This is only sent once per check run

Reply:
Good point 👍